core-ui/src/components/Extensibility/helpers/index.js

```diff
@@ -82,6 +83,35 @@ export function createTemplate(api, namespace, scope) {
   return template;
 }
 
+export function getDefaultPreset(presets, emptyTemplate) {
+  if (!presets || !presets.length) return null;
+  const defaultPreset = presets.find(preset => preset.default === true);
+  return defaultPreset
+    ? merge({}, { value: emptyTemplate }, defaultPreset)
+    : null;
+}
+
+export function createPresets(resource, presets) {
+  if (!presets || !presets.length) return null;
+  let preparedPresets = [];
+  let hasDefaultPreset = false;
+
+  presets.forEach(preset => {
+    if (preset.default === true) hasDefaultPreset = true;
+    preparedPresets.push(merge({}, { value: resource }, preset));
+  });
+
+  if (!hasDefaultPreset) {
+    preparedPresets.unshift({
+      name: 'default',
```

Comment:
Please use translations here.

Reply:
Also would be great to make use of translations in naming presets - someone might want to give them more descriptive, translatable name.